### PR TITLE
 Update http_transport decode to handle 32bit types

### PR DESCRIPTION
**Pull request by hasAdamr**
http_transport decode now properly converts 32bit floats and int from
64bit types return from strconv. Fixes issue #39

This PR also handles uint32 and uint64 for http_transport decode and
client argument generation

Also updating .travis.yml to run tests ./deftree directory

### CODE REVIEW

File: gengokit/httptransport/httptransport.go [hunk 1]
```diff
@@ -181,6 +180,31 @@ func createDecodeConvertFunc(f Field) string {
 	return fmt.Sprintf(fType, f.LocalName, f.LocalName+"Str")
 }
 
+// createDecodeTypeCast creates a go string that casts a 64 bit type to a 32 bit type
+// as strconv.ParseInt and ParseFloat always return the 64 bit type
+func createDecodeTypeCast(f Field) string {
```
Comment (tunezaq) [inline gengokit/httptransport/httptransport.go]: missing some types https://golang.org/ref/spec#Numeric_types

```
uint8       the set of all unsigned  8-bit integers (0 to 255)
uint16      the set of all unsigned 16-bit integers (0 to 65535)
uint32      the set of all unsigned 32-bit integers (0 to 4294967295)
uint64      the set of all unsigned 64-bit integers (0 to 18446744073709551615)

int8        the set of all signed  8-bit integers (-128 to 127)
int16       the set of all signed 16-bit integers (-32768 to 32767)
int32       the set of all signed 32-bit integers (-2147483648 to 2147483647)
int64       the set of all signed 64-bit integers (-9223372036854775808 to 9223372036854775807)
```

Comment (hasAdamr) [reply gengokit/httptransport/httptransport.go]: We only support the types in this table
https://developers.google.com/protocol-buffers/docs/proto3#scalar

File: gengokit/httptransport/httptransport.go [hunk 2]
```diff
@@ -181,6 +180,31 @@ func createDecodeConvertFunc(f Field) string {
...
+	fType := ""
+	switch {
```
Comment (tunezaq) [inline gengokit/httptransport/httptransport.go]: Only need the int32 and float32 case in here. Everything else is default

File: gengokit/httptransport/httptransport.go [hunk 3]
```diff
@@ -95,6 +95,7 @@ func NewBinding(i int, meth *deftree.ServiceMethod) *Binding {
 		}
 		nField.GoType = gt
 		nField.ConvertFunc = createDecodeConvertFunc(nField)
+		nField.TypeCast = createDecodeTypeCast(nField)
```
Comment (tunezaq) [inline gengokit/httptransport/httptransport.go]: Please name TypeConversion or a bunch of gophers will attack you in your sleep.

File: gengokit/clientarggen/client_argument_generator.go [hunk 4]
```diff
@@ -142,12 +142,14 @@ func New(svc *deftree.ProtoService) *ClientServiceArgs {
 func createFlagConvertFunc(a ClientArg) string {
 	fType := ""
 	switch {
+	case strings.Contains(a.GoType, "uint32"):
+		fType = "%s = flag.Uint(\"%s\", 0, %s)"
```
Comment (tunezaq) [inline gengokit/clientarggen/client_argument_generator.go]: Could use a raw string here

```
`%s = flag.Uint("%s", 0, %s)`
```

File: gengokit/clientarggen/client_argument_generator.go [hunk 5]
```diff
@@ -168,12 +170,14 @@ func createFlagConvertFunc(a ClientArg) string {
...
+	case strings.Contains(a.GoType, "uint64"):
+		fType = "*%s"
 	case strings.Contains(a.GoType, "int32"):
 		fType = "int32(*%s)"
 	case strings.Contains(a.GoType, "int64"):
 		fType = "*%s"
-	case strings.Contains(a.GoType, "int"):
-		fType = "*%s"
 	case strings.Contains(a.GoType, "bool"):
 		fType = "*%s"
```
Comment (tunezaq) [inline gengokit/clientarggen/client_argument_generator.go]: Could default these cases.

**Comment by tunezaq** [comment]
LGTM, 2 small nit pick things you could fix.

**Pull request merged**